content/tutorials/history.textile

```diff
@@ -78,6 +128,30 @@ h2. Step 4 - Publishing messages to be stored with channel history
 
 Before we can use the history API to retrieve previously published messages, we need to publish some messages to a channel. In the example code below, we publish 3 messages to the channel.
 
+blang[java].
+  ```[java]
+  public class ExampleActivity extends AppCompatActivity {
+    ...
+    private Channel channel;
+    ...
+    private void initAbly() throws AblyException {
+        ...
+        channel = ablyRest.channels.get("persisted:sounds");
+    }
+
+    private void publishMessages() throws AblyException {
+      if (channel != null) {
```

Comment:
Why `if (channel != null)`. Under what conditions would this happen and surely we should throw an exception instead of silently moving on?